cloudflare_worker/src/lib.rs

```diff
@@ -90,14 +91,14 @@ pub async fn create_signed_exchange(
 ) -> Result<JsValue, JsValue> {
     let payload_headers = ::sxg_rs::headers::Headers::new(payload_headers.into_serde().unwrap());
     let signer = Box::new(::sxg_rs::signature::js_signer::JsSigner::new(signer));
-    let sxg = WORKER.create_signed_exchange(::sxg_rs::CreateSignedExchangeParams {
+    let sxg: HttpResponse = WORKER.create_signed_exchange(::sxg_rs::CreateSignedExchangeParams {
         fallback_url: &fallback_url,
         now: std::time::UNIX_EPOCH + std::time::Duration::from_secs(now_in_seconds as u64),
         payload_body: &payload_body,
         payload_headers,
         signer,
         status_code,
-    }).await;
+    }).await.map_err(|err| JsValue::from_str(&err))?;
```

Comment:
@antiphoton when I remove this `.map_err(...)`, the compiler doesn't complain, but when I remove the one in `validate_payload_headers`, it complains. Any idea why that is?

Reply:
No idea. I don't see any difference between here and `validate_payload_headers`.

Reply:
Found it: https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator

> If the value is `Err(e)`, then it will return `Err(From::from(e))`